### PR TITLE
Get GitHub repos functional/ added jest tests/ added eslint

### LINKED ISSUE #2: Adding a repository
#2 **Add repos from GitHub into ADR-Manager**
As a user, I want to load public and private repositories into the ADR Manager so that I can edit (#24, #7) the ADR files in them.

**Acceptance criteria**
- [x]  Solving problem with CORS policy
- [x]  Request GitHub to give ADR-Manager **authorization** to get repos of user
- [ ]  Get and display data from the user

### PR DESCRIPTION
**Pull request by manuelmerkel**
# Description

- A backend has been set up which accepts requests and returns an answer. 
- It was set up eslint and jest. 

Fixes #2 
## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/pullRequestTemplate.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Description
```
Comment (koppor) [inline docs/pullRequestTemplate.md:1]: Typically, the filename is `pull_request_template.md`, isn't it? https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository

File: README.md [hunk 2]
```diff
@@ -15,11 +15,22 @@ npm run serve
 npm run build
 ```
 
-### Lints and fixes files
+### Run lint
```
Comment (koppor) [inline README.md:18]: Please use [markdown-lint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) to ensure consistent markdown styling 😇

File: backend/main.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+from flask import Flask, request
+from flask_cors import cross_origin
+import os
+import requests
+from flask.json import jsonify
+import json
+
+app = Flask(__name__)
+
+# This information is obtained upon registration of a new GitHub OAuth
```
Comment (koppor) [inline backend/main.py:11]: Should be received by an environment variable.

File: docs/pullRequestTemplate.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+Please delete options that are not relevant.
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+# Checklist:
+
```
Comment (koppor) [inline docs/pullRequestTemplate.md:17]: Provide a hint what do to with undo times. 

   Leave the checkbox blank if not done

File: docs/pullRequestTemplate.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+- [ ] My code follows the style guidelines of this project
```
Comment (koppor) [inline docs/pullRequestTemplate.md:18]: Link the guidelines

File: docs/pullRequestTemplate.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+## Type of change
+
+Please delete options that are not relevant.
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+# Checklist:
```
Comment (koppor) [inline docs/pullRequestTemplate.md:16]: Is the checklist really suitable for your project? Isn't it too hard?

Where is the entry "Change added to `CHANGELOG.md`"? For CHANGELOG.md see https://keepachangelog.com/ and https://github.com/JabRef/jabref/blob/master/CHANGELOG.md for a real-life example.

File: package.json [hunk 7]
```diff
@@ -3,14 +3,15 @@
     "version": "0.1.0",
     "private": true,
     "scripts": {
-        "serve": "vue-cli-service serve",
+        "serve": "vue-cli-service serve --open",
         "build": "vue-cli-service build",
-        "lint": "vue-cli-service lint"
```
Comment (koppor) [inline package.json:8]: Why was linting removed? The checklist included "no new warnings"

File: src/components/TheEditor.vue [hunk 8]
```diff
@@ -57,90 +57,90 @@
...
 
   </v-card>
-          
+
 </template>
 
 <script>
-  import _ from 'lodash'
-  import { md2adr, adr2md, ArchitecturalDecisionRecord } from '@/plugins/utilities'
+import lodash from 'lodash';
```
Comment (koppor) [inline src/components/TheEditor.vue:64]: Doesn't have lodash the convetoin to use `_`? (Hence the name low dash?) --> Should be commented, shoulddn't it? Even if a linter complains, just state that.

File: src/views/Register.vue [hunk 9]
```diff
@@ -12,14 +12,13 @@
...
 <script>
 // @ is an alias to /src
-  import GitLogIn from '@/components/UsefulComponents/GitLogIn.vue'
+import GitLogIn from '@/components/UsefulComponents/GitLogIn.vue';
 
 export default {
   name: 'Editor',
   components: {
-    GitLogIn
+    GitLogIn,
```
Comment (koppor) [inline src/views/Register.vue:20]: Really, a final comma at the end?

File: src/views/EditorView.vue [hunk 10]
```diff
@@ -41,57 +44,75 @@
...
+      this.adrRaw = adr2md(new ArchitecturalDecisionRecord());
+    },
+    addRepositories() {
+      axios
+          .post(this.reposPath)
+          .then((res) => {
+            console.log(res.data);
+          })
+          .catch((error) => {
+          // eslint-disable-next-line
```
Comment (koppor) [inline src/views/EditorView.vue:103]: Why? Because of the indent or because of the error reporting?

**Review by koppor — COMMENTED** [review]
Some minor comments.

I hope, there is some ADR saying why we need a backend and why it is written in python. Maybe, it is just an intermediate solution before we switch to some function-as-a-service hoster.

The backend should be hosted on Heroku, Digital Ocean or Azure in a way as cheap as possible. Not on some virtual server at Hetzner or somehwere elese.

**Comment by koppor** [comment]
Missing checklist item:

    - [ ] Screenshots added for changes

File: src/views/Register.vue (hunk 9 above)
Comment (manuelmerkel) [reply src/views/Register.vue:20]: Is deleted. Unfortunately prettier added automatically.

File: src/components/TheEditor.vue (hunk 8 above)
Comment (manuelmerkel) [reply src/components/TheEditor.vue:64]: I did't know that there was a conventions. Was changed again.

File: src/views/Register.vue (hunk 9 above)
Comment (xJREB) [reply src/views/Register.vue:20]: Yeah, I also had some issues in the past in VSCode to get Prettier to respect the rules I set for TSLint/ESLint... :D Especially about those trailing commas...

**Pull request merged**